src/index.ts

```diff
@@ -12,45 +10,169 @@ export async function init(code: BufferSource) {
     codec = await WebAssembly.instantiate(wasm, {})
 }
 
+// Returns true iff library was initialized with wasm code
 export function isInitialized() {
     return (codec !== null)
 }
 
+// User input to the encodeSpawnData function
+export interface SpawnData {
+    version: number,
+    template: string,
+    name: string
+    ctor_name: string,
+    calldata: Uint8Array,
+}
+
+// User input to the encodeCallData function
+export interface CallData {
+    version: number,
```

Comment:
basically, the `version` should always be 0 for now
do we want to have that part of the `interface`?

Reply:
@avive